static/js/services/select2-search.js

```diff
@@ -96,6 +97,22 @@ angular.module('inboxServices').factory('Select2Search',
           });
       };
 
+      var getDoc = function(id) {
+        return LineageModelGenerator.contact(id).then(function(model) {
+          var current = model.doc;
+          model.lineage.forEach(function(hydrated) {
```

Comment:
this should be done by the `LineageModelGenerator`, with a `getHydratedDoc` func.

Reply:
and a test for the `getHydratedDoc` func! 
(basically what I've been doing in sentinel 😞 )

Reply:
I've avoided doing that because I don't want to modify the doc in any way so that we can still save it later and get to use the optimistic locking features of couchdb.

Because this is being backported I've kept it simple. When we do #3826 we can make it more awesomer.

Reply:
then make the function return both or something. Code duplication for tricky algorithms != optimistic lock problem

Reply:
I don't really get what the issue is. The LineageModelGenerator calls a hydrate function behind the scenes and returns both (just like you suggested) a doc and a lineage. All this code does is merge them together. I think I'm doing what you suggested?

Reply:
After talking with Estelle we agreed to let this slide for this issue, and fix it up in #3826